### PR TITLE
Add missing crypto and s3 extras

### PR DESCRIPTION
**Pull request by potiuk**
I noticed that we are missing two extras that were existing
in Airflow 1.10 - crypto and s3. They are both deprecated
because s3 should be replaced with amazon and crypto is
already installed by default by Airflow as 'install_requires',
but for backwards compatibility, we should add them to 2.0 as well
with deprecation comment.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]

**Comment by potiuk** [comment]
Random faiulure, but pushing again just in case and also to run all testts.

**Comment by mik-laj** [comment]
I am not sure about the crypto change. If we use extra, which does not exist, the installation is still successful.

**Comment by mik-laj** [comment]
The cryptography package should now be required by the core, so I don't know if we need to repost the required versions for it.

**Comment by mik-laj** [comment]
On latest master:
```
13:04 $ pip install -e .[crypto]
.....
Requirement already satisfied: cryptography>=0.9.3 in /Users/kamilbregula/.virtualenvs/airflow/lib/python3.6/site-packages (from apache-airflow==2.0.0b3) (3.2.1)
....
Successfully installed apache-airflow

```

File: docs/apache-airflow/extra-packages-ref.rst [hunk 1]
```diff
@@ -208,6 +208,8 @@ Here's the list of the :ref:`subpackages <installation:extra_packages>` and what
 +=====================+=====================================================+======================================================================+
 | cgroups             | ``pip install 'apache-airflow[cgroups]'``           | Needed To use CgroupTaskRunner                                       |
 +---------------------+-----------------------------------------------------+----------------------------------------------------------------------+
+| crypto              | ``pip install 'apache-airflow[crypto]'``            | Used for cryptography operations                                     |
```
Comment (mik-laj) [inline docs/apache-airflow/extra-packages-ref.rst]: I think it would be useful to know that this option is deprecated and not needed in Airflow 2.0

Comment (potiuk) [reply docs/apache-airflow/extra-packages-ref.rst]: I 've already added and issue for that: https://github.com/apache/airflow/issues/12851. Let'\s move the conversation there if you have some ideas how to do it.

**Comment by potiuk** [comment]
> The cryptography package should now be required by the core, so I don't know if we need to repost the required versions for it.

I believe it is needed because of fernet encryption which is part of the core. So core absolutely need this package. The '[crypto]" extra I added is only there for the backwards-compatibility. I just want to avoid the situations where our users get random failures when they try to install Airflow the same way they did before. 

See https://github.com/apache/airflow/issues/12824 - the user wanted to install airflow with

`[async,crypto,celery,jdbc,password,postgres,s3]'

And when they tried to do it with Airflow 2.0 (I did) they would get a "missing extras" warning. Maybe that is enough in many cases but I would like to limit the number of issues people open "Hey it does not work in Airflow 2.0" and we agreed before that we do not remove but deprecate everything we can. 

Let's discuss ideas on how to do it in #12851, but i think we cannot simply just "drop" an extra without any explanation or earlier warning.

File: docs/apache-airflow/extra-packages-ref.rst (hunk 1 above)
Comment (potiuk) [reply docs/apache-airflow/extra-packages-ref.rst]: I also modified slightly the pre-commit now to be able to remove crypto from the check, and I indeed left an "empty' crypto requirement.

I also added a new table of deprecated extras. in the doc.

**Comment by potiuk** [comment]
I updated thedocs and removed the 'cryptography' from crypto (I made exclusion there) , I also removed yandexcloud as extra - it was not present in 1.10 at all so better to keep just 'yandex' in 2.0 in-sync with the provider name.

**Comment by potiuk** [comment]
And added list of deprecated extras for those who actually read the documentation.

**Pull request merged**